src/onnx/parse_convolution.cpp

```diff
@@ -149,9 +159,45 @@ struct parse_convolution : op_parser<parse_convolution>
 
         recalc_conv_attributes(values, kdims);
 
+        instruction_ref ret;
+        // parse a_zero_point and b_zero_point values
+        auto l0_zp         = set_bias(l0, 2, args);
+        auto w_zp          = set_bias(weights, 3, args);
+        auto is_quant_conv = opd.op_name == "quant_convolution";
+
         op.from_value(values);
-        auto l1 = info.add_instruction(op, l0, args[1]);
-        return info.add_bias(args, l1, 1);
+
+        // Check for type mismatch on parse
+        if(l0_zp->get_shape().type() != l0_shape.type())
+        {
+            MIGRAPHX_THROW("PARSE:Conv Data and Data Zero Point must have same type");
+        }
+
+        if(is_quant_conv && (w_zp->get_shape().type() != w_shape.type()))
+        {
+            MIGRAPHX_THROW("PARSE:ConvInteger Weights and Weights Zero Point must have same type");
+        }
+
+        // Assign input bias on weights or data accordingly
+        if(is_quant_conv && (l0_zp != l0))
```

Comment:
The idea of comparing L_0 to L_0_zp is not logical. We end up here because of set_bias(), above -- because of it,  L_0 compares to itself when no bias is present, at lines 171  -- or at 176. 
Suggestion: 1. Please separate the error handling, maybe as in https://github.com/ROCm/AMDMIGraphX/blob/77177af25ea1280af552f761293a352fa1c6e0f0/src/onnx/parse_qlinearmatmul.cpp#L96, 
Suggestion: 2. Remove comparisons like 'l0_zp != l0', as they can be equivalently handled by directly checking for index > 2 or index > 3 --- as that is all that set_bias() does!

Reply:
Done